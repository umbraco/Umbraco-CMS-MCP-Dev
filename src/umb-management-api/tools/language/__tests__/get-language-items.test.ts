import GetLanguageItemsTool from "../get/get-language-items.js";
import { LanguageBuilder } from "./helpers/language-builder.js";
import { LanguageTestHelper } from "./helpers/language-helper.js";
import { jest } from "@jest/globals";

describe("get-language-items", () => {
  const TEST_LANGUAGE_NAME = "_Test Language Items";
  const TEST_LANGUAGE_ISO = "en-GB";
  let originalConsoleError: typeof console.error;
  let builder: LanguageBuilder;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
    builder = new LanguageBuilder();
  });

  afterEach(async () => {
    await builder.cleanup();
    await LanguageTestHelper.cleanup(TEST_LANGUAGE_ISO);
    console.error = originalConsoleError;
  });

  it("should get language item by isoCode", async () => {
    // Ensure the test language exists
    await builder
      .withName(TEST_LANGUAGE_NAME)
      .withIsoCode(TEST_LANGUAGE_ISO)
      .withIsDefault(false)
      .withIsMandatory(false)
      .create();

    const result = await GetLanguageItemsTool().handler({ isoCode: [TEST_LANGUAGE_ISO] }, { signal: new AbortController().signal });
    expect(result).toMatchSnapshot();
  });

  it("should handle non-existent isoCode", async () => {
    const result = await GetLanguageItemsTool().handler({ isoCode: ["xx-NOTFOUND"] }, { signal: new AbortController().signal });
    expect(result).toMatchSnapshot();
  });
}); 