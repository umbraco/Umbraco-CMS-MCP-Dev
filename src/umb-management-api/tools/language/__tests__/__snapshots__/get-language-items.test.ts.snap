// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-language-items should get language item by isoCode 1`] = `
{
  "content": [
    {
      "text": "[{"name":"_Test Language Items","isoCode":"en-GB"}]",
      "type": "text",
    },
  ],
}
`;

exports[`get-language-items should handle non-existent isoCode 1`] = `
{
  "content": [
    {
      "text": "[]",
      "type": "text",
    },
  ],
}
`;
